app/controllers/TaxYearSummaryController.scala

```diff
@@ -168,7 +168,20 @@ class TaxYearSummaryController @Inject()(taxYearSummaryView: TaxYearSummary,
             documentDetail => DocumentDetailWithDueDate(documentDetail, documentDetail.getDueDate(),
               dunningLock = financialDetails.dunningLockExists(documentDetail.transactionId)))
         }
-        f(documentDetailsWithDueDates ++ documentDetailsWithDueDatesForLpi ++ documentDetailsWithDueDatesCodingOutPaye ++ documentDetailsWithDueDatesCodingOut)
+        val documentDetailZeroBalancingCharge: List[DocumentDetailWithDueDate] = {
+          documentDetails.filter(_.isBalancingCharge(isEnabled(CodingOut))).filter(_.originalAmount == Some(BigDecimal(0))).map {
```

Comment:
Should fix: originalAmount filtering is model logic, moving to isBalancingCharge in DocumentDetail would ensure all relevant logic is in one place. See topmost methods in DocumentDetail for similar examples